coderd/database/queries/oauth2.sql

```diff
@@ -60,3 +63,71 @@ WHERE id = $1 RETURNING *;
 
 -- name: DeleteOAuth2ProviderAppSecretByID :exec
 DELETE FROM oauth2_provider_app_secrets WHERE id = $1;
+
+-- name: GetOAuth2ProviderAppCodeByID :one
+SELECT * FROM oauth2_provider_app_codes WHERE id = $1;
+
+-- name: GetOAuth2ProviderAppCodeByAppIDAndSecret :one
+SELECT * FROM oauth2_provider_app_codes WHERE app_id = $1 AND hashed_secret = $2;
+
+-- name: InsertOAuth2ProviderAppCode :one
+INSERT INTO oauth2_provider_app_codes (
+    id,
+    created_at,
+    expires_at,
+    hashed_secret,
+    app_id,
+    user_id
+) VALUES(
+    $1,
+    $2,
+    $3,
+    $4,
+    $5,
+    $6
+) RETURNING *;
+
+-- name: DeleteOAuth2ProviderAppCodeByID :exec
+DELETE FROM oauth2_provider_app_codes WHERE id = $1;
+
+-- name: DeleteOAuth2ProviderAppCodesByAppAndUserID :exec
+DELETE FROM oauth2_provider_app_codes WHERE app_id = $1 AND user_id = $2;
+
+-- name: InsertOAuth2ProviderAppToken :one
+INSERT INTO oauth2_provider_app_tokens (
+    id,
+    created_at,
+    expires_at,
+    hashed_secret,
+    app_secret_id,
+    api_key_id
+) VALUES(
+    $1,
+    $2,
+    $3,
+    $4,
+    $5,
+    $6
+) RETURNING *;
+
+-- name: GetOAuth2ProviderAppsByUserID :many
+SELECT COUNT(DISTINCT oauth2_provider_app_tokens.id) as token_count,
+  sqlc.embed(oauth2_provider_apps)
+FROM oauth2_provider_app_tokens
+  INNER JOIN oauth2_provider_app_secrets
+    ON oauth2_provider_app_secrets.id = oauth2_provider_app_tokens.app_secret_id
+  INNER JOIN oauth2_provider_apps
+    ON oauth2_provider_apps.id = oauth2_provider_app_secrets.app_id
+  INNER JOIN api_keys
+    ON api_keys.id = oauth2_provider_app_tokens.api_key_id
+WHERE api_keys.user_id = $1
+GROUP BY oauth2_provider_apps.id;
```

Comment:
```suggestion
-- name: GetOAuth2ProviderAppsByUserID :many
SELECT
    COUNT(DISTINCT oauth2_provider_app_tokens.id) as token_count,
    sqlc.embed(oauth2_provider_apps)
FROM oauth2_provider_app_tokens
  INNER JOIN oauth2_provider_app_secrets
    ON oauth2_provider_app_secrets.id = oauth2_provider_app_tokens.app_secret_id
  INNER JOIN oauth2_provider_apps
    ON oauth2_provider_apps.id = oauth2_provider_app_secrets.app_id
  INNER JOIN api_keys
    ON api_keys.id = oauth2_provider_app_tokens.api_key_id
WHERE
    api_keys.user_id = $1
GROUP BY
    oauth2_provider_apps.id;
```

Formatting?

Reply:
My brain dislikes the hanging keywords, but I think that is just a matter of what I am used to and I do not have strong feelings.  Should we be enforcing this via the formatter?

Reply:
Formatted!  Actually this does look better.